index.js

```diff
@@ -96,10 +96,25 @@ function Hook (modules, options, onrequire) {
       }
       moduleName = filename
     } else {
-      const stat = parse(filename)
+      let stat = parse(filename)
       if (stat === undefined) {
-        debug('could not parse filename: %s', filename)
-        return exports // abort if filename could not be parsed
+        if (path.isAbsolute(filename) && hasWhitelist === true) {
+          const parsedPath = path.parse(filename)
+          if (modules.includes(filename) || modules.includes(path.join(parsedPath.root, parsedPath.dir, parsedPath.name))) {
+            stat = {
```

Comment:
This stat object doesn't match to what `module-details-from-path` returns usually. For example `resolveModuleName(stat)` below returns a `fullModuleName` which is incorrect.
Actually this `fullModuleName` is not needed in case of absolute requires so maybe skip the creation of this object and directly assign the values to `moduleName` and `basedir` and move the few lines below into the `!absoluteRequire` block.

Reply:
[`e2024fd` (#1)](https://github.com/dynatrace-oss-contrib/require-in-the-middle/pull/1/commits/e2024fd4215bdbc9a8524a644a9b56ad4776553b)